File: ReadIt/ViewModels/BlogModel.cs
namespace ReadIt.ViewModels
{
    public class BlogModel
    {
        public long Id { get; set; }

        public string Title { get; set; }

        public string Description { get; set; }

        public string Tags { get; set; }

        public long CreatedBy { get; set; }
        public UserModel User { get; set; }
        public List<CommentModel> Comments { get; set; }
        public long CategoryId { get; set; }
        public string CategoryName { get; set; }
        public DateTime CreatedOn { get; set; }
        public IFormFile? BlogImage { get; set; }
        public string BlogImageUrl { get; set; }
    }
}

File: ReadIt.Angular/src/app/core/models/blog.model.ts
import { CommentModel } from "./comment.model"
import { UserModel } from "./user.model"

export class Blog {
  id!: number
  title!: string
  description!: string
  tags!: string
  createdBy!: number
  categoryId!: number
  user!: UserModel
  totalComments?: number
  categoryName?: string
  createdOn?: Date
  blogImage?: File | null
  blogImageUrl?: string
}
